requirements.txt

```diff
@@ -7,3 +7,5 @@ polling
 -e git+https://github.com/ebi-ait/ingest-client.git#egg=hca_ingest
 six #flatten_json undeclared dependancy
 json-converter
+submission-broker
```

Comment:
I believe we should have concrete version numbers in the requirements file.

Reply:
to do this via the SOP I should use pip-compile:
Which throws the following error for ingest-client:
```
    WARNING: git clone in ./venv/src/hca-ingest exists with URL https://github.com/HumanCellAtlas/ingest-client.git
    WARNING: The plan is to install the git repository https://github.com/ebi-ait/ingest-client.git
      ERROR: Command errored out with exit status 1:
       command: /Users/astaffer/Repos/ingest-archiver/venv/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/dp/q6sgzxz97r9dsfy4vmr5v05nptwv5z/T/pip-resolver-997uc3ta/hca/setup.py'"'"'; __file__='"'"'/private/var/folders/dp/q6sgzxz97r9dsfy4vmr5v05nptwv5z/T/pip-resolver-997uc3ta/hca/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/dp/q6sgzxz97r9dsfy4vmr5v05nptwv5z/T/pip-pip-egg-info-fhd8b81p
           cwd: /private/var/folders/dp/q6sgzxz97r9dsfy4vmr5v05nptwv5z/T/pip-resolver-997uc3ta/hca/
      Complete output (3 lines):
      
      This repository has been deprecated. For information about accessing Human Cell Atlas data please refer to the `Data Browser quick start guide. https://data.humancellatlas.org/guides/quick-start-guide`
      
      ----------------------------------------
```
Should I go and fix ingest-client first?

Reply:
I am not sure I understand.
3rd party dependencies should have concrete version numbers, at least to the minor version level.
Regarding ingest-client, I am not sure I understand why we are based on the repo rather than on a concrete version.

Reply:
Ingest client is not a 3rd party dependancy, it's internal.
Removing the reference to ingest client allows pip-compile to work, so I have checked in the results, but we should fix this asap.